erpnext/schools/doctype/student_group/student_group.py

```diff
@@ -10,47 +10,64 @@
 from erpnext.schools.api import get_student_batch_students
 
 class StudentGroup(Document):
-	def autoname(self):
-		self.name = frappe.db.get_value("Course", self.course, "course_abbreviation")
-		if not self.name:
-			self.name = self.course
-		if self.student_batch:
-			self.name += "-" + self.student_batch
-		else:
-			prog_abb = frappe.db.get_value("Program", self.program, "program_abbreviation")
-			if not prog_abb:
-				prog_abb = self.program
-			if prog_abb:
-				self.name += "-" + prog_abb
-			if self.academic_year:
-				self.name += "-" + self.academic_year
-		if self.academic_term:
-			self.name += "-" + self.academic_term
-	
 	def validate(self):
+		self.validate_mandatory_fields()
 		self.validate_strength()
-		self.validate_student_name()
-		self.validate_name()
-		if self.student_batch:
-			self.validate_student_batch()
+		self.validate_students()
 		validate_duplicate_student(self.students)
 
+	def validate_mandatory_fields(self):
+		if self.group_type == "Course" and not self.course:
+			frappe.throw(_("Please select Course"))
+		elif self.group_type == "Batch" and (not self.program or not self.batch):
+			frappe.throw(_("Please select Program and Batch"))
+
 	def validate_strength(self):
 		if self.max_strength and len(self.students) > self.max_strength:
 			frappe.throw(_("""Cannot enroll more than {0} students for this student group.""").format(self.max_strength))
 
-	def validate_student_name(self):
-		for d in self.students:
-			d.student_name = frappe.db.get_value("Student", d.student, "title")
-	
-	def validate_name(self):
-		if frappe.db.exists("Student Batch", self.name):
-			frappe.throw(_("""Student Batch exists with same name"""))
-
-	def validate_student_batch(self):
-		student_batch_students = []
-		for d in get_student_batch_students(self.student_batch):
-			student_batch_students.append(d.student)
+	def validate_students(self):
+		program_enrollment = self.get_program_enrollment()
+		students = [d.student for d in program_enrollment] if program_enrollment else None
 		for d in self.students:
-			if d.student not in student_batch_students:
-				frappe.throw(_("""Student {0}: {1} does not belong to Student Batch {2}""".format(d.student, d.student_name, self.student_batch)))
\ No newline at end of file
+			if d.student not in students:
+				frappe.throw(_("{0} - {1} is not enrolled in the given {2}".format(d.student, d.student_name, self.group_type)))
+			if not frappe.db.get_value("Student", d.student, "enabled") and d.active:
+				d.active = 0
+				frappe.throw(_("{0} - {1} is inactive student".format(d.student, d.student_name)))
+
+
+	def update_students(self):
+		enrolled_students = self.get_program_enrollment()
+		group_student_list = [student.student for student in self.students]
+
+		if enrolled_students:
+			student_list = [];
+			for s in enrolled_students:
+				if s.student not in group_student_list:
+					student_list.append(s.update({"active": 1}) if frappe.db.get_value("Student", s.student, "enabled")
```

Comment:
This type of `val...if...else` conditional statement is better if it is not too long. Better to use separate if...else.